### ISSUE TITLE
Unable to compile ipfilter source code

### ISSUE THREAD

**Issue by akhilesh2410** [aixoss/ipfilter#5]
I was looking at the compilation steps for ipfilter source code and found that Makefile presents in git repo are not sufficient enough to compile source code. It has dependencies on some kernel source/library/modules(not known to me) as i saw VPATH is set with some /bos/kernel/exp path.

Also there are some Makefile which are missing like "bos/kernext/Kernext.mk" and "RULES_MK".,

Could you please educate me how can i get these files and compile source code ?

-bash-4.3# ipf -V
ipf: IP Filter: v4.1.13 (480)
Kernel: IP Filter: v4.1.13
Running: yes
Log Flags: 0 = none set
Default: pass all, Logging: available
Active list: 0
Feature mask: 0x87
-bash-4.3#

Thanks,
Akhilesh Verma

**Comment by davemq** [comment]
Yes, I've been a bit surprised when some customers have told me they've compiled this code. They must have made some changes to make it work.

I recently compiled some of the code, and changed some Makefiles to get things built. I'll create a pull request for this and see if anyone wants to review it.

You might also look at issue #1 about removing ADE includes from the Makefiles.

**Comment by akhilesh2410** [comment]
Thanks Dave,
Looked at the issue #1 it talks about the same.
JFYI,I also able to compile kernext by modifying existing Makefile,but it would be great if i can have Makefiles to compile entire source code.

**Makefile for kernext also need to update to build the source for kernext.**

Thanks,
Akhilesh Verma @davemq

**Comment by davemq** [comment]
If you'd like I can give you access to this Github repository so you can create your own branches and pull requests. It appears you've made the relevant changes, so you just need a way to share them.

**Comment by akhilesh2410** [comment]
It would be good if you can give me access but i am still working on it as it is showing some warnings,
Anyway will create a pull request with my changes.

Thanks,
Akhilesh Verma

**Comment by akhilesh2410** [comment]
@davemq  Please look into issue #4 as well, i have found and suggested some changes in code, if you can look at it and provide your input on the same that would be helpful.
